source/parser/x86_64/classifiers.hpp

```diff
@@ -187,10 +211,19 @@ namespace smeagle::x86_64 {
 
   inline classification classify(st::typeArray *t) {
     const auto size = t->getSize();
+
     if (size > 64) {
       return {RegisterClass::MEMORY, RegisterClass::NO_CLASS, "Array"};
     }
-    return {RegisterClass::INTEGER, RegisterClass::NO_CLASS, "Array"};
+
+    // Just classify the base type
+    const auto baseType = t->getBaseType();
+    auto [underlying_type, ptr_cnt] = unwrap_underlying_type(baseType);
+
+    if (ptr_cnt > 0) {
+      return classify_pointer(ptr_cnt);
+    }
+    return classify_type(underlying_type);
```

Comment:
I would go ahead and push the unwrapping and pointer check into the generic classifier since it's the same everywhere, as well.

Reply:
I changed it, but now there is a segfault - do you want me to push to a separate branch so you can take a look?